app/models/Campaign.coffee

```diff
@@ -34,4 +35,17 @@ module.exports = class Campaign extends CocoModel
     }
     sum = (nums) -> _.reduce(nums, (s, num) -> s + num) or 0
     stats.playtime = sum((session.get('playtime') or 0 for session in sessions))
-    return stats
\ No newline at end of file
+    return stats
+  
+  getLevels: ->
+    levelObjects = _.values(@get('levels'))
+    sorted = _.sortBy levelObjects, (level) ->
+      level.campaignIndex
+    return new Levels(sorted)
+    
+  getNonladderLevels: ->
+    levels = new Levels(_.values(@get('levels')))
+    levels.reset(levels.reject (level) -> level.isLadder())
+    levels.comparator = (level) -> level.get('campaignIndex')
```

Comment:
If [comparator](http://backbonejs.org/#Collection-comparator) is a string, it sorts by that attribute. So switch to:

`levels.comparator = 'campaignIndex'`